### ISSUE TITLE
Lower mAP when using smaller batch size during RetinaNet training?

### ISSUE THREAD

**Issue by BorisLestsov** [facebookresearch/detectron2#868]
## ❓ Lower mAP when using smaller batch size during RetinaNet training?

Hi,

I was able to train RetinaNet with default config `configs/COCO-Detection/retinanet_R_50_FPN_1x.yaml` and reproduce the result (my metric is 36.5322 mAP on COCO val set). However, when I try to train the same model but with smaller batch size, I get only 35.9391 mAP on the same COCO val set.

The difference in training configs `configs/Base-RetinaNet.yaml`:

Original config:
> SOLVER:
  IMS_PER_BATCH: 16
  BASE_LR: 0.01  # Note that RetinaNet uses a different default learning rate
  STEPS: (60000, 80000)
  MAX_ITER: 90000

Small batch size config:

> SOLVER:
  IMS_PER_BATCH: 8
  BASE_LR: 0.005  # Note that RetinaNet uses a different default learning rate
  STEPS: (120000, 160000)
  MAX_ITER: 180000

In both cases the models were trained on 8 GPUs.

I divided learning rate by 2 and multiplied number of iterations and STEPS parameters by 2, because the batch size is twice as small. Am I doing something wrong? 

I decided not to submit this issue as bug, because am probably doing something wrong, so it's not the issue of the repo.

**Comment by ppwwyyxx** [comment]
The changes look reasonable and I would also expect such changes to produce similar AP -- but there is no guarantee on how these hyperparameters affect results because this is deep learning :)

**Comment by ppwwyyxx** [comment]
There is an improvement of RetinaNet in https://github.com/facebookresearch/detectron2/commit/d362da6302ff1d96279183d4f6aa83274c53f19f that will make it perform as good as the model zoo in your setting. Thanks for bringing this up.

(we trust linear learning rate scaling rule and when it does not apply we look for potential issues in the model.
